v3/round_trip_with_retry_backoff.go

```diff
@@ -0,0 +1,41 @@
+package heroku
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/cenkalti/backoff"
+)
+
+func RoundTripWithRetryBackoff(req *http.Request) (*http.Response, error) {
+	var lastResponse *http.Response
+	var lastError error
+
+	retryableRoundTrip := func() error {
+		lastResponse, lastError = http.DefaultTransport.RoundTrip(req)
```

Comment:
probably need to check lastError and return it if not nol?

Reply:
`lastError` is checked just a few lines down [in the parent function](#diff-5c57c0fa1ac47d01d237b2e53639943aR33), because this `backoff.Retry` function should only return an error when it should be retried (status 429 for this rate-limiting use-case).

Reply:
You could dry it up by dropping `lastError` and doing something like this:

```
retryableRoundTrip := func() err error {
		lastResponse, err = http.DefaultTransport.RoundTrip(req)
		// Detect Heroku API rate limiting
		// https://devcenter.heroku.com/articles/platform-api-reference#client-error-responses
		if lastResponse.StatusCode == 429 {
			err = fmt.Errorf("Heroku API rate limited: 429 Too Many Requests")
		}
		return err
	}
```

That would also mean that you would only need to handle the error once below...

Reply:
But we only want this to retry for HTTP status **429** responses.

If that function returns other errors, it will retry for them too.

We only want to retry when rate-limited, right?

Reply:
Oh yeah 😊